osf/models/preprint_provider.py

```diff
@@ -128,8 +128,22 @@ def rules_to_subjects(rules):
             q.append(models.Q(_id=sub))
     return Subject.objects.filter(reduce(lambda x, y: x | y, q)) if len(q) > 1 else (Subject.objects.filter(q[0]) if len(q) else Subject.objects.all())
 
-
 @receiver(post_save, sender=PreprintProvider)
 def create_provider_auth_groups(sender, instance, created, **kwargs):
     if created:
         GroupHelper(instance).update_provider_auth_groups()
+
+
+class SharePreprintProviderWhitelisted(models.Model):
+    id = models.AutoField(primary_key=True)
+    provider_name = models.CharField(default=None, blank=True, unique=True, max_length=200)
+
+    @classmethod
+    def create(cls, provider_name):
+        provider, created = SharePreprintProviderWhitelisted.objects.get_or_create(provider_name=provider_name)
+        if created:
+            provider.save()
```

Comment:
If `provider` was created, it's already saved.

I think it would make more sense to delete `SharePreprintProviderWhitelisted.create` and instead use `SharePreprintProviderWhitelisted.objects.get_or_create`, since they now do exactly the same thing, and the latter is more djangoey/consistent.